django/utils/decorators.py

```diff
@@ -45,8 +45,18 @@ def _dec(obj):
         else:
             func = obj
 
+        # helper function to manage the iterable decorator case
+        def decorate(function):
+            if hasattr(decorator, "__iter__"):
+                # at this point we don't know if decorator respects
+                # the sequence protocol so we generate a tuple out of it
+                for dec in reversed(tuple(decorator)):
```

Comment:
Decided to generate a tuple and then reverse it.

The tuple is generated each time the decorate function is called.